### PR TITLE
[new] Netpune apex Integration

### PR DESCRIPTION
**Pull request by platinummonkey**
### What does this PR do?

Adds the Neptune Apex Controller integration. This collects metrics from your apex controller, an aquarium monitor.

Image 1-26-2023 at 4 26 PM

### Motivation

Monitoring my aquarium.

### Review checklist

- [x] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [x] Feature or bugfix has tests
- [x] Git history is clean
- [x] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: neptune_apex/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+### Events
+
+Neptune Apex does not include any events.
+
+## Troubleshooting
+
+Need help? Contact [Datadog support][3].
+
+[1]: **LINK_TO_INTEGRATION_SITE**
```
Comment (estherk15) [inline neptune_apex/README.md:52]: Link is left as placeholder

File: neptune_apex/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+3. [Download the Datadog Agent][2].
+
+4. Upload the build artifact to any host with an Agent and
+ run `datadog-agent integration install -w
+ path/to/neptune_apex/dist/<ARTIFACT_NAME>.whl`.
+
+### Configuration
+
+1. <List of steps to setup this Integration>
```
Comment (estherk15) [inline neptune_apex/README.md:28]: Configuration steps are left as placeholder

File: neptune_apex/README.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+### Validation
+
+<Steps to validate integration is functioning as expected>
```
Comment (estherk15) [inline neptune_apex/README.md:32]: Validations is left as placeholder

**Review by estherk15 — COMMENTED** [review]
Created a card for documentation to review [[DOCS-4768](https://datadoghq.atlassian.net/browse/DOCS-4768)]. Noticed that some sections were left as the placeholder value for your awareness.

File: neptune_apex/README.md (hunk 1 above)
Comment (platinummonkey) [reply neptune_apex/README.md]: added

File: neptune_apex/README.md (hunk 3 above)
Comment (platinummonkey) [reply neptune_apex/README.md]: done

File: neptune_apex/README.md (hunk 2 above)
Comment (platinummonkey) [reply neptune_apex/README.md]: done

**Comment by platinummonkey** [comment]
it appears the build system is broken here. It's arguing with itself on what is included and what is failing. I'm pretty blocked at this point. 

The errors aren't helpful and it isn't pointing to useful information.

**Comment by platinummonkey** [comment]
The `validations` job seems to argue with other Ci jobs about `metrics_to_check`. So I'm blocked, please fix this. There are also no other manifests that have that key.

`PR Changes` just outright fails without any helpful message at all.  Which again looks to be a base build & CI issue on your end. This is also blocking.

File: neptune_apex/CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# CHANGELOG - Neptune Apex
+
+## v0.0.1
+
+- Introduction to Neptune Apex Support
```
Comment (urseberry) [inline neptune_apex/CHANGELOG.md:5]:
```suggestion
- Introduction to Neptune Apex support
```

File: neptune_apex/README.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+3. [Download the Datadog Agent][2].
+
+4. Upload the build artifact to any host with an Agent and
+ run `datadog-agent integration install -w
+ path/to/neptune_apex/dist/<ARTIFACT_NAME>.whl`.
+
+### Configuration
+
+1. Edit your `/etc/datadog-agent/conf.d/neptune_apex.d/conf.yaml` and add your entry to your in-network apex controller.
+2. You will at a minimum need to specify the `address` which should be generally `http://your_apex_controller_ip`
```
Comment (urseberry) [inline neptune_apex/README.md:29]:
```suggestion
2. At a minimum, specify the `address`, which is generally `http://your_apex_controller_ip`.
```

File: neptune_apex/README.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+### Validation
+
+1. First ensure your `address` is correct. You can use curl or a browser and load `http://your_apex_controller_ip/cgi-bin/status.xml`. This should load an xml document.
```
Comment (urseberry) [inline neptune_apex/README.md:33]:
```suggestion
1. First, ensure your `address` is correct. You can use `curl` or a browser and load `http://your_apex_controller_ip/cgi-bin/status.xml`. This should load an xml document.
```

File: neptune_apex/README.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+2. Use `datadog-agent status` and search for the `neptune_apex` status line to indicate any errors collecting.
+
+## Data Collected
+
+### Metrics
+
+Neptune Apex includes several metrics on your available probes and outlets. This includes watts and amperes for the 120V outlets or any 
+device that draws power. For alarms additionally this includes the "status" if the alarm is configured.
```
Comment (urseberry) [inline neptune_apex/README.md:41]:
```suggestion
Neptune Apex includes several metrics on your available probes and outlets. This includes watts and amperes for 120V outlets or any 
device that draws power. For alarms, an additional metric of the "status" is available if the alarm is configured.
```

File: neptune_apex/README.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+### Service Checks
+
+Neptune Apex includes one service check which is the ability to connect to the device. This is useful to monitor if your whole apex controller suddenly becomes offline.
```
Comment (urseberry) [inline neptune_apex/README.md:45]:
```suggestion
Neptune Apex includes one service check, which is the ability to connect to the device. Monitor this service check to receive an alert if your whole Apex controller suddenly goes offline.
```

File: neptune_apex/README.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+### Events
+
+Neptune Apex includes outlet events. This happens if you turn outlets on or off.
```
Comment (urseberry) [inline neptune_apex/README.md:49]:
```suggestion
Neptune Apex includes outlet events. Outlet events happen if you turn outlets on or off.
```

**Review by urseberry — APPROVED** [review]

**Comment by platinummonkey** [comment]
y'all.... this is a bit crazy with the checks, the feedback cycle hasn't been good. I'm closing this and it can be an artifact if someone wants to use it

**Pull request closed without merging**